yucca/training/data_loading/YuccaDataset.py

```diff
@@ -11,21 +11,21 @@
 class YuccaTrainDataset(torch.utils.data.Dataset):
     def __init__(
         self,
-        preprocessed_data_dir: list,
+        samples: list,
         patch_size: list | tuple,
         keep_in_ram: Union[bool, None] = None,
         label_dtype: Optional[Union[int, float]] = None,
         composed_transforms=None,
-        task_type: Literal["classification", "segmentation", "unsupervised"] = "segmentation",
+        task_type: Literal["classification", "segmentation", "unsupervised", "contrastive"] = "segmentation",
     ):
-        self.all_cases = preprocessed_data_dir
+        self.all_cases = samples
         self.composed_transforms = composed_transforms
         self.patch_size = patch_size
         self.task_type = task_type
         if label_dtype is None:
             if self.task_type in ["segmentation", "classification"]:
                 self.label_dtype = torch.int32
-            if self.task_type == "unsupervised":
+            if self.task_type in ["unsupervised", "contrastive"]:
                 self.label_dtype = torch.float32
```

Comment:
I think this is super confusing and cannot understand why we need it, but code is OOP, so cannot understand how not either.